File: common/MechanicalWear/MechanicalWear.py
# Function to calculate efficiency: E(t) = E0 * (1 - t / T)
def calculate_efficiency(E0, T, t):
    return E0 * (1 - t / T)


# Function to find the minimum time t such that E(t) < E_target
def find_wear_time(E0, T, E_target):
    if E0 <= 0 or T <= 0 or E_target <= 0:
        raise ValueError("Initial efficiency, total life, and target efficiency must be positive.")

    t = 0  # Start time from 0
    E_t = E0  # Initial efficiency

    # Iteratively calculate until E(t) is less than the target efficiency
    while E_t >= E_target:
        t += 0.1  # Increment time by 0.1
        E_t = calculate_efficiency(E0, T, t)

    return t  # Return the minimum time t


# Main function to get user input and calculate the wear time
def main():
    # Get user input
    E0 = float(input("请输入初始效率 E0: "))
    T = float(input("请输入设备总寿命 T: "))
    E_target = float(input("请输入目标效率 E_target: "))

    # Find the minimum wear time
    time = find_wear_time(E0, T, E_target)

    # Output the result
    print(f"设备的效率将小于 {E_target} 在时间 t = {time} 时发生。")


# Run the main function
if __name__ == "__main__":
    main()

File: common/MechanicalWear/MechanicalWear.java
//有一个机械装置，随着使用时间的增加，装置的效率会下降。可以通过一个公式来模拟其磨损，并计算出需要多少时间才能达到设定的效率水平
import java.util.Scanner;

public class MechanicalWear {

    // 计算磨损后的效率公式 E(t) = E0 * (1 - t / T)
    public static double calculateEfficiency(double E0, double T, double t) {
        return E0 * (1 - t / T);
    }

    // 找到最小的时间 t，使得 E(t) < E_target
    public static double findWearTime(double E0, double T, double E_target) {
        if (E0 <= 0 || T <= 0 || E_target <= 0) {
            throw new IllegalArgumentException("Initial efficiency, total life, and target efficiency must be positive.");
        }

        double t = 0;  // 时间从0开始
        double E_t = E0;  // 初始效率

        // 迭代计算直到 E(t) 小于目标值
        while (E_t >= E_target) {
            t += 0.1;  // 逐步增加时间
            E_t = calculateEfficiency(E0, T, t);
        }

        return t;  // 返回最小时间 t
    }

    public static void main(String[] args) {
        // 获取用户输入
        Scanner scanner = new Scanner(System.in);
        System.out.print("请输入初始效率 E0: ");
        double E0 = scanner.nextDouble();

        System.out.print("请输入设备总寿命 T: ");
        double T = scanner.nextDouble();

        System.out.print("请输入目标效率 E_target: ");
        double E_target = scanner.nextDouble();

        // 找到最小时间 t
        double time = findWearTime(E0, T, E_target);

        // 输出结果
        System.out.println("设备的效率将小于 " + E_target + " 在时间 t = " + time + " 时发生。");
    }
}
